app/actions/remote/entry/app.ts

```diff
@@ -35,16 +38,13 @@ export const appEntry = async (serverUrl: string, since = 0) => {
     const rolesData = await fetchRoles(serverUrl, teamData?.memberships, chData?.memberships, meData?.user, true);
 
     if (initialTeamId === currentTeamId) {
-        let cId = await getCurrentChannelId(database);
         if (tabletDevice) {
-            if (!cId) {
-                const channel = await getDefaultChannelForTeam(database, initialTeamId);
-                if (channel) {
-                    cId = channel.id;
-                }
+            const cId = await getNthLastChannelFromTeam(database, currentTeamId);
```

Comment:
ok this entire code block needs to change.

Here we do the following

the initial team is the first team we still belong to and in this case it means that we still belong to the current team, so this part of the code is alright. Where does it need to change, in the else statement as the initial team can be a team that we already visited and it could probably be that we switched teams while in the global threads.

So I would change this block to the following:

```ts
    let cId = await getNthLastChannelFromTeam(database, initialTeamId);
    if (tabletDevice) {
        if (!cId) {
            const channel = await getDefaultChannelForTeam(database, initialTeamId);
            if (channel) {
                cId = channel.id;
            }
        }

        if (cId === Screens.GLOBAL_THREADS) {
            setCurrentTeamAndChannelId(operator, initialTeamId, '');
            switchToGlobalThreads(serverUrl);
        } else {
            switchToChannelById(serverUrl, cId, initialTeamId);
        }
    } else {
        setCurrentTeamAndChannelId(operator, initialTeamId, '');
    }
```

basically we get the last channel we visited for the initialTeam which can return an empty string if no channel has been visited regardless of being a tablet or not.
Then in case of a tablet if no channel has been visited we get the default channel.
If the last visited channel of the initial team, not just the current one is the global threads we set the initialTeam as current and no current channel and finally switch to it the global thread view, if not, we just switch to the channel.

then if it is not a tablet, we just set the initial team and leave the channel as blank.

let me know if you have questions about this.

Reply:
I'll return to this after https://github.com/mattermost/mattermost-mobile/pull/6139 is reviewed and merged

Reply:
https://github.com/mattermost/mattermost-mobile/blob/1132bd0e510d0c83b65f65d246670ae9a3c4f2cb/app/queries/servers/team.ts#L83-L89

`getNthLastChannelFromTeam` is already returning the default channel id I